.github/workflows/deploy-sdk.yml

```
@@ -0,0 +1,193 @@
+                  }
+                },
+                {
+                  "type": "divider"
+                },
+                {
+                  "type": "section",
+                  "text": {
+                    "type": "mrkdwn",
+                    "text": "*Version ${{ steps.semantic-release.outputs.new_release_version }}*\n\nReactNative SDK deployment progress:\n ~1. <https://github.com/${{github.repository}}/releases/tag/${{steps.semantic-release.outputs.new_release_version}}|create git tag>~\n2. deploy to npm\n\n"
+                  }
+                }
+              ]
+            }
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+          SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
+
+      - name: Notify team of failure
+        uses: slackapi/slack-github-action@v1.18.0
+        if: ${{ failure() }} # only run this if any previous step failed
+        with:
+          # Use block kit to format the Slack message: https://app.slack.com/block-kit-builder
+          payload: |
+            {
+              "text": "ReactNative SDK deployment failure",
+              "username": "ReactNative deployment bot",
+              "icon_url": "https://vectorified.com/images/icon-react-native-24.png",
+              "channel": "#mobile-deployments",
+              "blocks": [
+                {
+                  "type": "section",
+                  "text": {
+                    "type": "mrkdwn",
+                    "text": "*ReactNative* SDK deployment :warning: failure :warning:"
+                  }
+                },
+                {
+                  "type": "divider"
+                },
+                {
+                  "type": "section",
+                  "text": {
+                    "type": "mrkdwn",
+                    "text": "ReactNative SDK failed deployment during step *create git tag*. View <https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}|CI server logs> to learn why and fix the issue. <https://github.com/customerio/mobile/blob/main/GIT-WORKFLOW.md|Learn more about the deployment process and how to fix errors>."
+                  }
+                }
+              ]
+            }
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+          SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
+
+  deploy-npm:
+    name: Deploy to npm
+    needs: [deploy-git-tag]
+    if: needs.deploy-git-tag.outputs.new_release_published == 'true' # only run if a git tag was made.
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+          cache: 'yarn'
+      - run: yarn install --frozen-lockfile
+
+      - name: Deploy to npm
+        run: ./scripts/deploy-code.sh
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          IS_PRERELEASE: ${{ github.event.release.prerelease }} # true or false boolean if a prerelease github release
```

```suggestion
          IS_PRERELEASE: false # at this time, all deployments are made to `main` production branch
```

Since `github.event.release` will no longer work, this logic will need updated. 

I [made a note](https://github.com/customerio/issues/issues/9287#issuecomment-1426189931) in another ticket which will address this feature.

---

Gotcha. I missed this one :(

If we do want prereleases would it make sense to keep a `beta` branch here? https://github.com/customerio/customerio-reactnative/pull/92/files#diff-862eb72ff2100b1cf86b7b1e618d58fcbe2ba9a5657ff05a577f9540b86b59d8R7